go_lib/bassa.go

```diff
@@ -0,0 +1,738 @@
+
+	request, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+	return string(out)
+}
+
+// GetUserSignupRequests : Function to get user signup requests
+func (b *Bassa) GetUserSignupRequests() string {
+
+	endpoint := "/api/user/requests"
+	apiURL := b.apiURL + endpoint
+
+	request, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+	return string(out)
+}
+
+// ApproveUserRequest : Function to approve user request
+func (b *Bassa) ApproveUserRequest(userName string) {
+	if userName == "" {
+		panic(errIncompleteParams)
+	}
+	endpoint := "/api/user/approve"
+	apiURL := b.apiURL + endpoint + "/" + userName
+
+	request, err := http.NewRequest("POST", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+}
+
+// GetBlockedUserRequests : Function to get blocked user requests
+func (b *Bassa) GetBlockedUserRequests() string {
+
+	endpoint := "/api/user/blocked"
+	apiURL := b.apiURL + endpoint
+
+	request, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+	return string(out)
+}
+
+// BlockUserRequest : Function to block user request
+func (b *Bassa) BlockUserRequest(userName string) {
+	if userName == "" {
+		panic(errIncompleteParams)
+	}
+	endpoint := "/api/user/blocked"
+	apiURL := b.apiURL + endpoint + "/" + userName
+
+	request, err := http.NewRequest("POST", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
```

Comment:
err is ignored

Reply:
I assume you are prettifying your json here.. OK now I see why err is ignored.
However is it necessary to log prettified JSON? maybe just log response.Body?

Reply:
Yeah, I am prettifying JSON 😅😅. 

I agree I will handpick the JSON only with useful data and return it back instead of complete response ✌🏻